airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSource.java

```diff
@@ -6,39 +6,115 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.common.collect.ImmutableMap;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
 import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.StreamingJdbcDatabase;
 import io.airbyte.integrations.base.IntegrationRunner;
 import io.airbyte.integrations.base.Source;
 import io.airbyte.integrations.source.jdbc.AbstractJdbcSource;
+import java.io.IOException;
 import java.sql.JDBCType;
+import java.sql.SQLException;
+import java.util.Properties;
 import java.util.Set;
+import javax.sql.DataSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class SnowflakeSource extends AbstractJdbcSource<JDBCType> implements Source {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeSource.class);
+  private static final HikariConfig hikariConfig = new HikariConfig();
+  private static final SnowflakeAccessTokenLoader tokenLoader = new SnowflakeAccessTokenLoader(hikariConfig);
   public static final String DRIVER_CLASS = "net.snowflake.client.jdbc.SnowflakeDriver";
+  public static boolean isSourceAlive;
 
   public SnowflakeSource() {
-    super(DRIVER_CLASS, new SnowflakeJdbcStreamingQueryConfiguration(), new SnowflakeSourceOperations());
+    super(DRIVER_CLASS, new SnowflakeJdbcStreamingQueryConfiguration(),
+        new SnowflakeSourceOperations());
   }
 
   public static void main(final String[] args) throws Exception {
     final Source source = new SnowflakeSource();
     LOGGER.info("starting source: {}", SnowflakeSource.class);
+    isSourceAlive = true;
     new IntegrationRunner(source).run(args);
+    isSourceAlive = false;
+    tokenLoader.interrupt();
     LOGGER.info("completed source: {}", SnowflakeSource.class);
   }
 
   @Override
-  public JsonNode toDatabaseConfig(final JsonNode config) {
+  public JdbcDatabase createDatabase(JsonNode config) throws SQLException {
+    final DataSource dataSource = createDataSource(config);
+    var database = new StreamingJdbcDatabase(dataSource, new SnowflakeSourceOperations(),
+        new SnowflakeJdbcStreamingQueryConfiguration());
+    quoteString = database.getMetaData().getIdentifierQuoteString();
+    return database;
+  }
+
+  private HikariDataSource createDataSource(final JsonNode config) {
```

Comment:
can this method be encapsulated in SnowflakeAccessTokenLoader?

Reply:
Done. Moved to `SnowflakeDataSourceUtils`